packages/webcrypto-ed25519-polyfill/src/secrets.ts

```diff
@@ -131,10 +131,54 @@ export function verifyPolyfill(key: CryptoKey, signature: BufferSource, data: Bu
     if (key.type !== 'public' || !key.usages.includes('verify')) {
         throw new DOMException('Unable to use this key to verify', 'InvalidAccessError');
     }
-    const publicKeyBytes = ed25519.getPublicKey(getSecretKeyBytes_INTERNAL_ONLY_DO_NOT_EXPORT(key));
+    const storedBytes = getSecretKeyBytes_INTERNAL_ONLY_DO_NOT_EXPORT(key);
     try {
-        return ed25519.verify(bufferSourceToUint8Array(signature), bufferSourceToUint8Array(data), publicKeyBytes);
+        return (
+            ed25519.verify(
+                bufferSourceToUint8Array(signature),
+                bufferSourceToUint8Array(data),
+                ed25519.getPublicKey(storedBytes)
+            ) || ed25519.verify(bufferSourceToUint8Array(signature), bufferSourceToUint8Array(data), storedBytes)
+        );
     } catch {
         return false;
     }
 }
+
+export function importKeyPolyfill(
+    format: KeyFormat,
+    keyData: BufferSource,
+    extractable: boolean,
+    keyUsages: readonly KeyUsage[]
+): CryptoKey {
+    const type =
+        // could be more rigorous
+        format === 'pkcs8' ? 'private' : 'public';
+
+    const bytes = bufferSourceToUint8Array(keyData);
+
+    const keyBytes =
+        format === 'pkcs8'
+            ? // Remove header
+              bytes.slice(16)
```

Comment:
I think a more robust way to do this without parsing the header would be to extract the last 32 bytes
We know that this is an ED25519 private key with a potentially variable length header, so we know the length of the key after the header
This assumes that the input is valid, while properly parsing the header would let us catch and report potential issues with it, but for now I think this would be sufficient

Reply:
Sorry for overruling you in a later comment. I think leaving the PKCS#8 payload unvalidated is wandering into dark territory. If we're going to sell this thing as a polyfill I want it to stick as close to the spec as possible.